### ISSUE TITLE
[Feature Request] Add more options to the "Open in" dropdown

### ISSUE THREAD

**Issue by cosimochellini** [ForkIssues/TrackerWin#936]
Hi! As a user, it would be incredibly useful for me to be able to do Open in => Open in webstorm.
I believe this setting can be very useful, even with all other jetbrains products like pycharm intelliJ and phpStorm.
Maybe even configurable from the settings 👼

**Comment by DanPristupov** [comment]
Does Webstorm use project files? How can I determine intelliJ, phpStorm projects?

**Comment by cosimochellini** [comment]
No, all JetBrains programs are pretty much the same as how **vs code** works, i.e. no project/solution files.
To open the project
[program name] .
(in my case `webstorm.bat .` )

Probably the most difficult thing is to detect the programs in use 😕  but maybe you can simply opt-in from the settings without proactively checking, you delegate the responsibility to the user to know that a certain program is correctly configured on his pc/mac

(from jebrains doc)
- https://www.jetbrains.com/help/webstorm/opening-files-from-command-line.html


edit: 
- maybe you can search for a **.idea** folder to enable all the jetbrains folder in the root directory of the project

**Comment by JeReT** [comment]
Same would be awesome for visual studio.
Not sure if you already have visual studio implemented. I work with Unity projects where all .sln / .csporj files are generated by the Unity Editor and therefore are in .gitignore.

I would prefer an option to configure used programs anyway. Sometimes I would like to open a file in Notepad++ also...

**Comment by DanPristupov** [comment]
> Same would be awesome for visual studio.

It's implemented.

> I would prefer an option to configure used programs anyway. Sometimes I would like to open a file in Notepad++ also...

You can create a custom command for that (File -> Preferences -> Custom Command. Click on +, 'Add Repository Custom Command')

**Comment by lubomirblazekcz** [comment]
Support for PhpStorm / IntelliJ would be nice.

Here is an example how it's implementend in Github Desktop - https://github.com/desktop/desktop/blob/development/app/src/lib/editors/win32.ts#L86
